src/utils/src/tempdir.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::env::temp_dir;
+use std::ops::Deref;
+use std::path::Path;
+
+use vmm_sys_util::{errno::Result, tempdir};
+
+/// Wrapper over vmm_sys_util::tempfile::TempDir struct.
+/// It provides methods to create create a temporary directory
+/// inside $TMPDIR if set, otherwise inside `/tmp`.
+pub struct TempDir(tempdir::TempDir);
+
+impl TempDir {
+    pub fn new() -> Result<Self> {
+        let mut in_tmp_dir = temp_dir();
+        // This `push` adds a trailing slash ("/tmp" -> "/tmp/").
+        // This is safe for paths with already trailing slash.
+        in_tmp_dir.push("");
+        let temp_dir = tempdir::TempDir::new(in_tmp_dir)?;
+        Ok(TempDir(temp_dir))
+    }
```

Comment:
I would've liked to have this directly in `vmm-sys-util` and not wrap it ourselved in Firecracker. I say you open a PR there as well and not let this dangling.

Reply:
I agree. We should expose this functionality in `vmm-sys-util`. I will open a PR there soon.